webhooks/tests/test_tasks.py

```diff
@@ -0,0 +1,95 @@
+
+from courses.factories import CourseFactory
+from webhooks.models import Webhook
+from webhooks.tasks import publish_webhook
+
+
+class PublishWebhookTests(TestCase):
+    """
+    Tests for webhook publishing.
+    """
+    def test_model_getting_catches_error(self):
+        """
+        Should catch error if invalid model to import.
+        """
+        try:
+            publish_webhook('foo.Bar', 1)
+        except LookupError:
+            self.fail("Didn't handle the lookup error")
+
+    def test_malformed_model_str_catches_error(self):
+        """
+        Should catch error if invalid model to import.
+        """
+        try:
+            publish_webhook('foo.Bar.Zot', 1)
+        except ValueError:
+            self.fail("Didn't handle malformed strings")
+
+    def test_model_posts_to_endpoint(self):
+        """
+        Happy case of posting.
+        """
+        Webhook.objects.create(url="http://example.org")
+        course = CourseFactory.create()
+        with mock.patch('webhooks.tasks.requests') as mock_requests:
+            publish_webhook('courses.Course', course.pk)
+            assert mock_requests.post.call_count == 1
+            args, _ = mock_requests.post.call_args
+            payload = json.loads(args[1])
+            assert payload['action'] == 'update'
+
+    def test_no_post_if_no_webhook_method(self):
+        """
+        If there's no to_webhook method, don't attempt to serialize.
+        """
+        Webhook.objects.create(url="http://example.org")
+        u = User.objects.create()
+        with mock.patch('webhooks.tasks.requests') as mock_requests:
+            publish_webhook('auth.User', u.pk)
+            assert mock_requests.post.call_count == 0
+
+    def test_no_model_sends_delete(self):
+        """
+        If there's no model, send a delete.
+        """
+        Webhook.objects.create(url="http://example.org")
+        with mock.patch('webhooks.tasks.requests') as mock_requests:
+            publish_webhook('courses.Course', 1)
+            assert mock_requests.post.call_count == 1
+            args, _ = mock_requests.post.call_args
+            payload = json.loads(args[1])
+            assert payload['action'] == 'delete'
+
+    def test_secure_header_verification(self):
+        """
+        Verifies that we sign each webhook request.
+        """
+        wh = Webhook.objects.create(url="http://example.org")
+        course = CourseFactory.create()
+        with mock.patch('webhooks.tasks.requests') as mock_requests:
+            publish_webhook('courses.Course', course.pk)
+            assert mock_requests.post.call_count == 1
+            args, kwargs = mock_requests.post.call_args
+            payload = json.loads(args[1])
+            assert payload['action'] == 'update'
+
+            # This constant_time_compare is important for implementations to
+            # ensure they're not vulnerable to timing attacks.
+            assert constant_time_compare(
```

Comment:
It's not useful in testing, aside from the fact that I can link this to someone as for the proper way to implement this validation on the client side.